extensions/Worker.Extensions.ServiceBus/src/ServiceBusSessionMessageActionsConverter.cs

```diff
@@ -49,7 +54,17 @@ public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
                     throw new InvalidOperationException("Expecting SessionLockedUntil within binding data of session actions and value was not present.");
                 }
 
-                var sessionActionResult = new ServiceBusSessionMessageActions(_settlement, sessionId!.ToString(), sessionLockedUntil.GetDateTimeOffset());
+                // Logic for if isBatched is true, then sessionIdArray will be used to get the sessionId.
+                var sessionIdRepeatedFieldArray = sessionIdArray as RepeatedField<string>;
+
+                if (foundSessionIdArray && (sessionIdRepeatedFieldArray == null || sessionIdRepeatedFieldArray.Count != 1))
```

Comment:
```suggestion
                if (foundSessionIdArray && (sessionIdRepeatedFieldArray is null || sessionIdRepeatedFieldArray.Count == 0))
```

Reply:
@liliankasem did you mean to approve? This feedback indicates there's a bug with the current logic.

Reply:
Meant to just submit, updated my review

Reply:
did not let me change it without a new comment so this is that ☠️

Reply:
changed to address feedback